transport/src/main/java/io/netty/channel/nio/SelectedSelectionKeySet.java

```diff
@@ -107,4 +62,25 @@ public boolean contains(Object o) {
     public Iterator<SelectionKey> iterator() {
         throw new UnsupportedOperationException();
     }
+
+    void reset() {
+        reset(0);
+    }
+
+    void reset(int start) {
+        for (; start < size; ++start) {
+            keys[start] = null;
+        }
+        size = 0;
```

Comment:
general use of these methods may result in non `null` entries being in the set when the size is zero, but keep in mind this class is package private and so are these methods. This shouldn't happen in practice based upon the narrow use case of this class (as we iterate we `null` elements out, and if we decide to prematurely stop iteration we call this method to `null` out the remaining elements https://github.com/netty/netty/pull/6083#discussion_r100674659).